src/main/java/gyro/google/compute/AbstractSslCertificateResource.java

```diff
@@ -0,0 +1,122 @@
+import gyro.core.GyroException;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+
+public abstract class AbstractSslCertificateResource extends ComputeResource implements Copyable<SslCertificate> {
+
+    private String certificatePath;
+    private String description;
+    private String name;
+    private String privateKeyPath;
+
+    // Read-only
+    private String selfLink;
+
+    /**
+     * The full path to the file containing the certificate(s) (in PEM format). The certificate chain must be no greater than 5 certs long. The chain must include at least one intermediate cert. Only used when creating the SSL certificate. (Required)
+     */
+    @Required
+    public String getCertificatePath() {
+        return certificatePath;
+    }
+
+    public void setCertificatePath(String certificatePath) {
+        this.certificatePath = certificatePath;
+    }
+
+    /**
+     * An optional description of this SSL certificate.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * The name of the SSL certificate. Must be 1-63 characters long, and the first character must be a lowercase letter. All other characters must be a lowercase letter, digit, or ``-``, except the last character, which cannot be a ``-``. (Required)
+     */
+    @Regex("[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?")
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The full path to the file containing the private key (in PEM format). Only used when creating the SSL certificate. (Required)
+     */
+    @Required
+    public String getPrivateKeyPath() {
+        return privateKeyPath;
+    }
+
+    public void setPrivateKeyPath(String privateKeyPath) {
+        this.privateKeyPath = privateKeyPath;
+    }
+
+    /**
+     * Server-defined URL for the SSL certificate.
+     */
+    @Id
+    @Output
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+
+    @Override
+    public void copyFrom(SslCertificate model) throws Exception {
+        setDescription(model.getDescription());
+        setName(model.getName());
+        setSelfLink(model.getSelfLink());
+    }
+
+    protected String readCertificateFile() {
+        try {
+            return new String(Files.readAllBytes(Paths.get(getCertificatePath())));
+        } catch (Exception ex) {
+            throw new GyroException("Failed reading certificate file: " + getCertificatePath());
+        }
+    }
+
+    protected String readPrivateKeyFile() {
+        try {
+            return new String(Files.readAllBytes(Paths.get(getPrivateKeyPath())));
```

Comment:
The `openInput()` utility method helps resolve relative path
So in the configs rather than providing the full path relative to the configuration the path could be set
```suggestion
           return IoUtils.toString(openInput(getPrivateKeyPath()), UTF_8);
```